crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -609,13 +661,18 @@ impl<'src> CodeExample<'src> {
     fn add_start(
         &mut self,
         original: InputDocstringLine<'src>,
-    ) -> Option<InputDocstringLine<'src>> {
+        queue: &mut VecDeque<CodeExampleAddAction<'src>>,
+    ) {
         assert!(self.kind.is_none(), "expected no existing code example");
         if let Some(doctest) = CodeExampleDoctest::new(original) {
             self.kind = Some(CodeExampleKind::Doctest(doctest));
-            return None;
+            queue.push_back(CodeExampleAddAction::Kept);
+        } else if let Some(litblock) = CodeExampleRst::new(original) {
+            self.kind = Some(CodeExampleKind::Rst(litblock));
+            queue.push_back(CodeExampleAddAction::Print { original });
+        } else {
+            queue.push_back(CodeExampleAddAction::Print { original });
```

Comment:
Does this mean that we'll push to the `VecDeque` even in case where the docstring contains no code exampels? I would be interested in understanding the performance implication of allocating here

Reply:
Yeah it will. Although it will push to the queue for every line, it should only allocate the first time. Even in the case where a code example is present, I don't expect the queue to ever grow beyond more than a couple elements, so the allocation should be amortized. But it's only amortized within each docstring.

It does indeed look like the happy path in the status quo does not allocate. I still perceive this to likely be a marginal cost, but I'll see about adding some benchmarks to this PR.